cozmo_echo.py

```diff
@@ -7,30 +7,35 @@
 
 def main():
 
-    
+    start_dialogue = False
+
     my_convo = Dialogue('speech.wav')
     my_convo.get_cozmo_response("Hello! I will echo back what you say!")
-
-    try:
-        print("starting recording process")
-        recorder = Recorder("speech.wav")
-        print("Please say something nice into the microphone\n")
-        recorder.save_to_file()
-        print("Transcribing audio....\n")
+    
+    while not start_dialogue:
 
         try:
-            speech_text = my_convo.transcribe_audio()
-            my_convo.get_cozmo_response(speech_text)
+            print("starting recording process")
+            recorder = Recorder("speech.wav")
+            print("Please say something nice into the microphone\n")
+            recorder.save_to_file()
+            print("Transcribing audio....\n")
 
-        except:
-            traceback.print_exc()
-            print("error in speech detection")
-            my_convo.get_cozmo_response("Sorry I couldn't understand you")
-            sys.exit(0)
+            try:
+                speech_text = my_convo.transcribe_audio()
+                my_convo.get_cozmo_response(speech_text)
+                if "goodbye" or "bye" in speech_text:
+                    break
+        
+            except:
```

Comment:
what exception?

Reply:
If there is some error either with recording the speech from the user or in the Speech to Text process, I want to raise exceptions. But that is where I am not sure what that exception needs to be, or how many. Something like a ValueError, or something directly provided by the Speech to Text Service, as noted here: https://cloud.ibm.com/apidocs/speech-to-text?code=python#error-handling.

Reply:
You would have to look at what exceptions the lines in the try block (`speech_text = my_convo.transcribe_audio()` and `my_convo.get_cozmo_response(speech_text)`) can possible raise and catch them.